packages/astro/src/runtime/server/render/instruction.ts

```diff
@@ -11,18 +11,32 @@ export type RenderHeadInstruction = {
 	type: 'head';
 };
 
+/**
+ * Render a renderer-specific hydration script before the first component of that
+ * framework
+ */
+export type RendererHydrationInstruction = {
+	type: 'renderer-hydration';
```

Comment:
You're right, the thing it's actually doing is rendering a hydration *script*, not doing the actual hydration. (Hydration happens after.) However, I feel that `hydration-script` would be confusingly similar to the unrelated `hasHydrationScript` thing. For the benefit of future developers I think it would be good to keep these clearly separated. I will rename it to `renderer-hydration-script` for now. Please take a look again after I push the next commit and let me know what you think?